vsifile/io/s3.py

```diff
@@ -45,120 +44,49 @@ def __hash__(self):
-                    aws_session_token=access_token,
-                    region_name=region_name or None,
-                )
-
+        parsed = urlparse(self.name)
+        bucket = parsed.netloc
+        self._key = parsed.path.strip("/")
+
+        config = {}
+        keys = {k.upper() for k in list(self.config)}
+        endpoint_url = os.environ.get("AWS_S3_ENDPOINT", None)
+        use_https = os.environ.get("AWS_HTTPS", "YES")
+        if not {"AWS_ENDPOINT_URL", "AWS_ENDPOINT"}.intersection(keys) and endpoint_url:
             # AWS_S3_ENDPOINT and AWS_HTTPS are GDAL config options of vsis3 driver
             # https://gdal.org/user/virtual_file_systems.html#vsis3-aws-s3-files
-            endpoint_url = os.environ.get("AWS_S3_ENDPOINT", None)
-            if endpoint_url:
-                use_https = os.environ.get("AWS_HTTPS", "YES")
-                if use_https.upper() in ["YES", "TRUE", "ON"]:
-                    endpoint_url = "https://" + endpoint_url
-
-                else:
-                    endpoint_url = "http://" + endpoint_url
-
-            self.client = session.client("s3", endpoint_url=endpoint_url)
-
-        parsed = urlparse(self.name)
-        self.bucket = parsed.netloc
-        self.key = parsed.path.strip("/")
-
-    def __enter__(self):
-        """Open file and fetch header."""
-        logger.debug(f"Opening: {self.name} (mode: {self.mode})")
-
-        head = self.client.head_object(Bucket=self.bucket, Key=self.key)
-        assert head["ResponseMetadata"]["HTTPStatusCode"] == 200
-        assert head.get("AcceptRanges") == "bytes"
-
-        self._mtime = int(head["LastModified"].timestamp())
-        self._size = int(head.get("ContentLength")) or 0
-        self.header = self._get_header()
+            if use_https.upper() in ["YES", "TRUE", "ON"]:
+                config["AWS_ENDPOINT_URL"] = "https://" + endpoint_url
+            else:
+                config["AWS_ENDPOINT_URL"] = "http://" + endpoint_url
+
+        options = {}
+        keys = [k.upper() for k in list(self.client_options)]
+        if "ALLOW_HTTP" not in keys and use_https in ["NO", "FALSE", "OFF"]:
+            options["ALLOW_HTTP"] = "TRUE"
+
+        region_name_env = (
+            os.environ.get("AWS_REGION", os.environ.get("AWS_DEFAULT_REGION")) or None
+        )
+
+        region_keys = [
+            "AWS_REGION",
+            "aws_region",
+            "region",
+            "AWS_DEFAULT_REGION",
+            "aws_default_region",
+            "default_region",
+        ]
+        region_name_config = next((config[k] for k in region_keys if k in config), None)
+
+        if self.infer_region and not region_name_config:
+            config["AWS_REGION"] = _find_bucket_region(bucket) or region_name_env
+
+        self._store = S3Store(
+            bucket,
+            config={**self.config, **config},
+            client_options={**self.client_options, **options},
+            retry_config=self.retry_config,
+        )
```

Comment:
in the last commit I remove the possibility of using boto3_session `S3Store.from_session` 

I think the main advantage was the possibility of using `AWS PROFILE`. I think this will be taken care or in obstore directly (or we will add it back here on user request)

Reply:
It should be handled in https://github.com/developmentseed/obstore/pull/203

Reply:
You can test this out with `0.4.0-beta.2`: https://developmentseed.org/obstore/v0.4.0-beta.2/api/store/aws/#obstore.store.S3Store._from_native

Reply:
I'm marking it as provisional for the 0.4.0 release